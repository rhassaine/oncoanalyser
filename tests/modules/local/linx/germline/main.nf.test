nextflow_process {
    name "Test Process LINX_GERMLINE"
    script "modules/local/linx/germline/main.nf"
    process "LINX_GERMLINE"
    config "tests/nextflow.config"

    test("LINX_GERMLINE with default parameters") {
        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/'
                resource_path = '/home/rhassaine/resources_files_oa/38/'
                mode = 'WGS'
            }
            process {
                """
                input[0] = [
                    [id: 'test', sample_id: 'subject_a.tumor'],
                    file('/home/rhassaine/container_test_output/subject_a/purple/subject_a.tumor.purple.sv.germline.vcf.gz')
                ]
                input[1] = 38
                input[2] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/common/ensembl_data/", checkIfExists: true)
                input[3] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/common/DriverGenePanel.38.tsv", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.annotation_dir != null },
                { assert process.out.annotation_dir.size() > 0 },
                { assert snapshot(process.out).match() }
            )

            if (process.success) {
                def linx_output_dir = process.out.annotation_dir[0][1]
                def sample_id = process.out.annotation_dir[0][0].sample_id
                assertAll(
                    { assert path(linx_output_dir).resolve("linx.version").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.germline.breakend.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.germline.clusters.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.germline.disruption.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.germline.driver.catalog.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.germline.links.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.germline.svs.tsv").exists() },
                    { assert process.out.versions != null },
                    { assert process.out.versions.size() > 0 },
                    { assert path(process.out.versions[0]).exists() }
                )
            }
        }
    }

    test("LINX_GERMLINE should fail when input is missing") {
        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/'
                resource_path = '/home/rhassaine/resources_files_oa/38/'
                mode = 'WGS'
            }
            process {
                """
                input[0] = []  // Missing input
                """
            }
        }

        then {
            assertAll(
                { assert !process.success },
                { assert process.exitStatus != 0 }
            )
        }
    }
}