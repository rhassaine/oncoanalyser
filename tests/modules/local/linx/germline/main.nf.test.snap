{
    "LINX_GERMLINE with default parameters": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "sample_id": "subject_a.tumor"
                        },
                        [
                            "linx.version:md5,a5fc97b5c4e6092096f0a1fdffd20159",
                            "subject_a.tumor.linx.germline.breakend.tsv:md5,62e0235c504dc8b2eb168d308ad1ca43",
                            "subject_a.tumor.linx.germline.clusters.tsv:md5,e73bc24fed69e59e14c7a0820b507d7b",
                            "subject_a.tumor.linx.germline.disruption.tsv:md5,02e772341ff55bd6c4ac3b33c71bc06c",
                            "subject_a.tumor.linx.germline.driver.catalog.tsv:md5,1ab1213e1408409affd036f247928194",
                            "subject_a.tumor.linx.germline.links.tsv:md5,b1d1360fa6598420a8d9751ba2a4d4dc",
                            "subject_a.tumor.linx.germline.svs.tsv:md5,ebba2c61f884b97eaf13be69817a0bed"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,c25d0eb259f72685d0f53a5127e91140"
                ],
                "annotation_dir": [
                    [
                        {
                            "id": "test",
                            "sample_id": "subject_a.tumor"
                        },
                        [
                            "linx.version:md5,a5fc97b5c4e6092096f0a1fdffd20159",
                            "subject_a.tumor.linx.germline.breakend.tsv:md5,62e0235c504dc8b2eb168d308ad1ca43",
                            "subject_a.tumor.linx.germline.clusters.tsv:md5,e73bc24fed69e59e14c7a0820b507d7b",
                            "subject_a.tumor.linx.germline.disruption.tsv:md5,02e772341ff55bd6c4ac3b33c71bc06c",
                            "subject_a.tumor.linx.germline.driver.catalog.tsv:md5,1ab1213e1408409affd036f247928194",
                            "subject_a.tumor.linx.germline.links.tsv:md5,b1d1360fa6598420a8d9751ba2a4d4dc",
                            "subject_a.tumor.linx.germline.svs.tsv:md5,ebba2c61f884b97eaf13be69817a0bed"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c25d0eb259f72685d0f53a5127e91140"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-20T04:39:55.098404267"
    }
}